### PR TITLE
Fixed and improved bad orders (accented, numerical and uppercase)

### PR DESCRIPTION
**Pull request by gusbemacbe**
Hello @jmredfern !

Before, your sort tool ordered incorrectly the accented letters, numbers (in reference to #1) and uppercase letters.

- Fixed and support for alphabetical order of accented, non-accented and uppercase letters;
- Support for numerical order: 0, 1, 2, 310, 11, 20, 22, 30, 33;
- Added support for semicolon, very used in the `.desktop` files of Linux apps.

"3; 2; 10; 40; 6; 4; 30; 33; 1; Gustavo; julho; Klaus; keyboard; último; árbol; uma; água; Argentina; Ángelo; argelino; unido" will become "1; 2; 3; 4; 6; 10; 30; 33; 40; água; árbol; ángelo; argelino; Argentina; Gustavo; keyboard; Klaus; julho; último; uma; unido;".

### CODE REVIEW

File: sort-selection/README.md [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+# Sort Selection
 
-This plugin adds the "Sort selection" command.
+This plugin adds the "Sort selection" and "Sort selection with semicolon" commands.
 
-The command will sort a comma delimited list within a selection.
+The command will sort a comma or semicolon delimited list within a selection.
 
-### Example:
+### Examplea:
```
Comment (jmredfern) [inline sort-selection/README.md:7]: typo, should be 
```suggestion
### Examples:
```

File: sort-selection/extension.js [hunk 2]
```diff
@@ -21,9 +21,29 @@ function activate(context) {
 }
 exports.activate = activate;
 
-function sort(txt) {
+// These codes are taken from https://stackoverflow.com/a/15479354, done by Georg Barikin (https://github.com/gebrkn)
+
+function naturalCompare(a, b) 
+{
+    var ax = [], bx = [];
```
Comment (jmredfern) [inline sort-selection/extension.js:28]: better to never use `var` but rather `const` or `let`

File: sort-selection/semicolon.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    return ax.length - bx.length;
+}
+
+function sort(txt) 
+{
+	const tokens = txt.split(/;\s*/);
+	const sortedTokens = tokens.sort(naturalCompare);
+	const sortedText = sortedTokens.join('; ');
```
Comment (jmredfern) [inline sort-selection/semicolon.js:50]: most of the functions in `semicolon.js` and `extension.js` are duplicated so it'd be nice to refactor them into a separate file :)

**Review by jmredfern — APPROVED** [review]
I left a few comments, if you wish to make the changes you're welcome, or if not I'll merge it as it is :)

File: sort-selection/extension.js (hunk 2 above)
Comment (gusbemacbe) [reply sort-selection/extension.js]: You are right. I had to keep myself updated with ES6. Done.

File: sort-selection/semicolon.js (hunk 3 above)
Comment (gusbemacbe) [reply sort-selection/semicolon.js]: I have removed this file, it was just a backup.

**Comment by gusbemacbe** [comment]
Done, @jmredfern.

**Comment by gusbemacbe** [comment]
Ah, I have to add `;` with `,` together. I'll refactor.

File: sort-selection/semicolon.js [hunk 4]
```diff
@@ -1,53 +0,0 @@
-const vscode = require('vscode');
-
-function activate(context) 
-{
-	console.log('"sort-selection-semicolon" is active!');
```
Comment (jmredfern) [inline sort-selection/semicolon.js]: Are you sure this file was intended to be removed? it looks like it provided the `sort-selection-semicolon` functionality.

By the way, my previous comment was that most of the functions are duplicated, the only differences are L7, L5 and L50. So I suggested to make a third file for the common code, and keep just the differences in these two files.

File: sort-selection/extension.js [hunk 5]
```diff
@@ -21,9 +21,29 @@ function activate(context) {
...
+
+function naturalCompare(a, b) 
+{
+    const ax = [], bx = [];
+
+    a.replace(/(\d+)|(\D+)/g, function(_, $1, $2) { ax.push([$1 || Infinity, $2 || ""]) });
+    b.replace(/(\d+)|(\D+)/g, function(_, $1, $2) { bx.push([$1 || Infinity, $2 || ""]) });
+    
+    while(ax.length && bx.length) {
+        var an = ax.shift();
```
Comment (jmredfern) [inline sort-selection/extension.js:34]: a few more vars here :)

File: sort-selection/semicolon.js (hunk 4 above)
Comment (gusbemacbe) [reply sort-selection/semicolon.js]: Yes, I created another extension for semicolon, that is mostly used for Linux desktop files like [BCompare desktop file](https://github.com/gusbemacbe/wine-desktop-files/blob/master/bcompare/bcompare.desktop). My goal is trying to unifying the comma and semicolon in the same only file `extension.js`

File: sort-selection/extension.js (hunk 5 above)
Comment (gusbemacbe) [reply sort-selection/extension.js:34]: Ah, I forgot. I reviewed quickly.

**Comment by gusbemacbe** [comment]
@jmredfern 

Ready! I tested and it worked. You can test with:

```
3; 2; 10; 40; 6; 4; 30; 33; 1; Gustavo; julho; Klaus; keyboard; último; árbol; uma; água; Argentina; Ángelo; argelino; unido

3, 2, 10, 40, 6, 4, 30, 33, 1, Gustavo, julho, Klaus, keyboard, último, árbol, uma, água, Argentina, Ángelo, argelino, unido

3 - 2 - 10 - 40 - 6 - 4 - 30 - 33 - 1 - Gustavo - julho - Klaus - keyboard - último - árbol - uma - água - Argentina - Ángelo - argelino - unido
```

**Review by jmredfern — APPROVED** [review]
good work! :+1:

**Pull request merged**